Analysis/ScanLibraries/source/XiaListModeDataMask.cpp

```diff
@@ -37,10 +36,13 @@ FIRMWARE XiaListModeDataMask::ConvertStringToFirmware(const std::string &type) {
         firmware = R30980;
     else if (firmwareNumber >= 30981 && firmwareNumber < 34688)
         firmware = R30981;
-    else if (firmwareNumber >= 34688)
+    else if (firmwareNumber >= 34688) {
         firmware = R34688;
-    else
-        throw invalid_argument("XiaListModeDataMask::CovnertStringToFirmware : Unrecognized firmware option - " + type);
+        if(firmwareNumber > 34688)
+            cout << "XiaListModeDataMask::ConvertStringToFirmware :  You requested a firmware number that's higher "
+                    "than the last known number. You should confirm that headers has the same format !!" << endl;
```

Comment:
This doesn't set the firmware to the latest value! It appears that `UNKNOWN` is returned.

Reply:
Also why not the following:
```c++
else if (firmwareNumber > 34688) {
   cout << "XiaListModeDataMask::ConvertStringToFirmware :  You requested a firmware number that's higher "
   "than the last known number. You should confirm that headers has the same format !!" << endl;
   firmware = R34688;
}
else
   throw invalid_argument("XiaListModeDataMask::ConvertStringToFirmware : Unrecognized firmware option - " + type)
```

Reply:
Are you reading that right? Line 40 sets the value, lines 41 - 43 send the message if needed and line 44 closes that if else block.

Reply:
No I wasn't reading it write. Leave it to git to leave to with an incomprehensible patch.

Reply:
You need to have the >= to handle the case for when the value is equal to 34688.

Reply:
I updated my code block to make it clear what I was thinking, but the point is mute as it all came out of a misunderstanding of your code. If you think my suggestion is cleaner feel free to incorporate it, but your solution is fine.

Reply:
I actually had your version at one point, but didn't want to repeat the code block to set the value. I'm thinking that we could replace this whole thing with a switch and really make it concise.

Reply:
Switch doesn't work because there are a bunch of in between values?

Reply:
You can have a fall through handle those.

Reply:
I don't think so:
```c++
void switch_test(int input) {
   std::cout << "Input: " << input << "\n";

   switch (input) {
      case 30980:
         std::cout << "Case 30980\n";
      case 34688:
         std::cout << "Case 34688\n";
      default:
         std::cout << "Didn't find a match!\n";
   }
}
```
Leads to this output:
```
$ root
root [0] .L switch_test.C 
root [1] switch_test(31234)
Input: 31234
Didn't find a match!
root [2] switch_test(30980)
Input: 30980
Case 30980
Case 34688
Didn't find a match!
root [3] 
```

Reply:
You are correct.